### PR TITLE
[RELAY]sch and compute for reduce ops

### PR DESCRIPTION
**Pull request by siju-samuel**
#2051
Thanks for contributing to TVM!   Please refer to guideline https://docs.tvm.ai/contribute/ for useful information and tips. After the pull request is submitted, please request code reviews from [Reviewers](https://github.com/dmlc/tvm/blob/master/CONTRIBUTORS.md#reviewers).

### CODE REVIEW

**Comment by jroesch** [comment]
Looks good to me once CI passes.

**Comment by siju-samuel** [comment]
```
======================================================================
ERROR: test_top_level4.test_reduce
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/local/lib/python2.7/dist-packages/nose/case.py", line 197, in runTest
    self.test(*self.arg)
  File "/workspace/nnvm/tests/python/compiler/test_top_level4.py", line 98, in test_reduce
    verify_reduce((4, 4, 3), np.sum, sym.sum)
  File "/workspace/nnvm/tests/python/compiler/test_top_level4.py", line 56, in verify_reduce
    verify_reduce_explicit(dshape, data, result, fsym, oshape=oshape, otype=otype, **kwargs)
  File "/workspace/nnvm/tests/python/compiler/test_top_level4.py", line 37, in verify_reduce_explicit
    graph, lib, _ = nnvm.compiler.build(y, target, {"x": dshape})
  File "/workspace/nnvm/python/nnvm/compiler/build_module.py", line 305, in build
    graph = graph.apply("GraphCompile")
  File "/workspace/nnvm/python/nnvm/graph.py", line 234, in apply
    check_call(_LIB.NNGraphApplyPasses(self.handle, npass, cpass, ctypes.byref(ghandle)))
  File "/workspace/nnvm/python/nnvm/_base.py", line 75, in check_call
    raise NNVMError(py_str(_LIB.NNGetLastError()))
NNVMError: [12:10:58] /workspace/nnvm/src/compiler/compile_engine.cc:212: Check failed: out[i].ndim() == out_info[i].ndim() (0 vs. 1) sum
```
Any idea why `nnvm/tests/python/compiler/test_top_level4.py` script is failing in CI?

File: src/relay/op/tensor/reduce.cc [hunk 1]
```diff
@@ -224,7 +238,19 @@ values over a given axis.
...
-.add_type_rel("ArgReduce", ArgReduceRel);
+.add_type_rel("ArgReduce", ArgReduceRel)
+.set_attr<FTVMCompute>(
+  "FTVMCompute", [](const Attrs& attrs,
+                    const Array<Tensor>& inputs,
+                    const Type& out_type,
+                    const Target& target) {
+    const auto* param = attrs.as<ReduceAttrs>();
+    auto r_axes = GetReduceAxes(inputs[0]->shape.size(),
+                                param->axis, param->exclude);
```
Comment (tqchen) [inline src/relay/op/tensor/reduce.cc]: This is certainly the shortest line of addition to implement the feature. But not necessarily the fastest to run. Note that the reduction in topi already supports negative index. So we don't need to handle negative index here, the only thing we need to handle is exclude, we can use another function GetExcludeAxes to get the other axes and return them as Array<Integer> when there is exclude

File: python/tvm/relay/op/_reduce.py [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+import topi
+from .op import  register_schedule, register_pattern
+from .op import schedule_injective, OpPattern
+
+schedule_broadcast = schedule_injective
+schedule_elemwise = schedule_injective
+
+def _schedule_reduce(_, outs, target):
+    """Generic schedule for reduce"""
+    with tvm.target.create(target):
```
Comment (tqchen) [inline python/tvm/relay/op/_reduce.py]: with target:

**Comment by tqchen** [comment]
see the compute impl in https://github.com/dmlc/tvm/pull/2163 that reflects the commend, the test-cases are still helpful and we would like to bring it in after 2163

**Pull request merged**